kafka_certificates/src/main/java/com/tesla/data/certificates/keystore/KafkaClientKeystores.java

```diff
@@ -75,17 +76,21 @@ public KeyStore createKeystore(KeystoreConfig conf) throws IOException, Certific
    * @return a keystore for the private key + chain of certificates
    */
   public KeyStore createKeystore(InputStream privateKey, InputStream certificate,
-      InputStream caChain) throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+                                 InputStream caChain) throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
     // initialize the keystore
     KeyStore ks = KeyStore.getInstance(JAVA_KEYSTORE);
     // need to load to initialize the keystore for use
     ks.load(null, password);
 
     // read the private key
     PEMParser parser = new PEMParser(new InputStreamReader(privateKey));
-    PrivateKeyInfo o = (PrivateKeyInfo) parser.readObject();
+    Object key = parser.readObject();
+    if (key instanceof PEMKeyPair) {
+      key = ((PEMKeyPair) key).getPrivateKeyInfo();
+    }
+
     JcaPEMKeyConverter converter = new JcaPEMKeyConverter().setProvider(BOUNCY_CASTLE_TYPE);
-    PrivateKey pk = converter.getPrivateKey(o);
+    PrivateKey pk = converter.getPrivateKey((PrivateKeyInfo) key);
```

Comment:
Does it make sense to assert key type is one between `PrivateKeyInfo` or `PEMKeyPair` and throw an exception if its anything but those two?